### ISSUE TITLE
IsADirectoryError in __init__

### ISSUE THREAD

**Issue by almaslov** [chrisjbillington/inotify_simple#17]
Hello!
Got an error when initializing new object. MWE:
```
import inotify_simple
print('Version', inotify_simple.__version__)
inotify_simple.INotify()
```
result is:
```
Version 1.3.3
Traceback (most recent call last):
  File "/tmp/in.py", line 4, in <module>
    inotify_simple.INotify()
  File ".../venv/lib/python3.5/site-packages/inotify_simple.py", line 86, in __init__
    FileIO.__init__(self, _libc_call(_libc.inotify_init1, flags), mode='rb')
IsADirectoryError: [Errno 21] Is a directory: 3
```
The problem as I understand it: 
FileIO checks stat flags for received file descriptor. We use legacy CentOS 6 with kernel v2.6.32 in production(is not up to me). inotify descriptor has S_IFDIR flag set, which means it is a directory.

Simple test indepenent from python and library version:
```
#include <sys/inotify.h>
#include <sys/stat.h>
#include <stdlib.h>
#include <stdio.h>


int main() {
  int fd = inotify_init();
  if (fd < 0) {
    perror("inotify_init");
    exit(1);
  }

  struct stat st;
  if (fstat(fd, &st) < 0) {
    perror("fstat");
    exit(1);
  }

  printf("0x%X\n", st.st_mode & S_IFMT);
  printf((st.st_mode & S_IFDIR) ? "DIR\n" : "NOT_DIR\n");
  return 0;
}
```
On 4.9.184 kernel problem is not reproduced.

**Comment by chrisjbillington** [comment]
Yikes, that's pretty annoying.

Changing `inotify_simple` to fix this would mean changing it to not be a subclass of `FileIO`. I've investigated other ways of doing it but nothing seems possible.

I don't think I want to do that to support an unsupported kernel. However, `inotify_simple` 1.2.1 (the last release before the change to subclass `FileIO`) will continue to work just fine - the only known bugs in that version are that [`it raises`](https://github.com/letorbi/synoindexwatcher/issues/34) `OSError: [Errno 0] Success` instead of `FileNotFoundError` when you try to watch a file that doesn't exist, and possible poor performance if using the `delay` argument to `INotify.read()`.

Could you simply pin the version at 1.2.1 if you want to use `inotfiy_simple` on old kernels where the inotify file descriptor is a directory?

**Comment by almaslov** [comment]
Yep, it's ok. Thank you.

**Comment by chrisjbillington** [comment]
Thanks for your understanding :)

If you do end up affected by the known bugs in 1.2.1, let me know and I can just push out a fixed 1.2.2.
